src/Core/Animation/KeyFramedValueInterpolators.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef RADIUMENGINE_KEYFRAMEDVALUE_INTERPOLATOR_HPP
+#define RADIUMENGINE_KEYFRAMEDVALUE_INTERPOLATOR_HPP
+
+#include <Core/Animation/KeyFramedValue.hpp>
+#include <Core/Animation/Pose.hpp>
+#include <Core/Animation/PoseOperation.hpp>
+#include <Core/Math/Interpolation.hpp>
+
+namespace Ra {
+namespace Core {
+namespace Animation {
+
+/**
+ * Look for the keyframes around time \p t.
+ * \param[in] keyframes The list of keyframes.
+ * \param[in] t The time to search for.
+ * \param[out] t0 The time for the keyframe preceding t.
+ * \param[out] t1 The time for the keyframe following t.
+ * \param[out] dt Linear parameter, \f$ \in [0;1] \f$, of \p t between \p t0 and \p t1.
+ * \note If \p t is lower than the first keyframe's time tf,
+ *       then \p t0 = \p t1 = tf and \p dt = 0.
+ *       If \p t is higher than the last keyframe's time tl,
+ *       then \p t0 = \p t1 = tl and \p dt = 0.
+ *       If \p t is an exact match on a keyframe's time tt,
+ *       then \p t0 = \p t1 = tt and \p dt = 0.
+ */
+template <typename T>
+void findRange( const typename KeyFramedValue<T>::KeyFrames& keyframes,
```

Comment:
I think this function should be moved in a class called `KeyFrames` (I would rename it as `KeyFrameCollection`), which would inherit `std::map` (would replace the `typedef` you have in `KeyFramedValue`) and add this specific lookup method.

Reply:
As of now, keyframes are stored in a vector, would you inherit from std::vector to add this method?